examples/vector_iteration.rs

```diff
@@ -1,14 +1,28 @@
+#[cfg(feature = "serde")]
+extern crate serde;
+
+#[cfg(not(feature = "serde"))]
 extern crate rustc_serialize;
 extern crate mustache;
 
 use std::str;
 use std::collections::HashMap;
 
-#[derive(RustcEncodable)]
+#[cfg_attr(not(feature = "serde"), derive(RustcEncodable))]
 struct User {
     name: String
 }
 
+#[cfg(feature = "serde")]
+impl serde::ser::Serialize for User {
```

Comment:
I think having to `include!` a code-genned file in example code is a bit confusing for people not using serde. (if someone already *is* interested in using serde they would find better ways to do this in serde's docs!)